frame/support/src/storage/types/mod.rs

```diff
@@ -42,9 +42,11 @@ pub use value::StorageValue;
 /// Trait implementing how the storage optional value is converted into the queried type.
 ///
 /// It is implemented by:
-/// * `OptionQuery` which convert an optional value to an optional value, user when querying storage
-///   will get an optional value.
-/// * `ValueQuery` which convert an optional value to a value, user when querying storage will get a
+/// * `OptionQuery` which converts an optional value to an optional value, used when querying
+///   storage returns an optional value.
+/// * `ResultQuery` which converts an optional value to a result value, used when querying storage
```

Comment:
Update this page too? https://paritytech.github.io/substrate/master/frame_support/attr.pallet.html

Reply:
worth mentioning that the error can only be pallet error (if that's correct, which I am only guessing?)